commons/src/main/java/com/ge/predix/acs/commons/web/BaseRestApiControllerAdvice.java

```diff
@@ -16,40 +16,106 @@
 
 package com.ge.predix.acs.commons.web;
 
-import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import org.springframework.beans.factory.annotation.Autowired;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.HttpMediaTypeNotAcceptableException;
+import org.springframework.web.HttpMediaTypeNotSupportedException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
-import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import com.ge.predix.acs.commons.exception.UntrustedIssuerException;
 
 @ControllerAdvice
 public class BaseRestApiControllerAdvice {
 
-    @Autowired
-    private RestErrorHandler restErrorHandler;
-
-    /**
-     * Annotated method that handles a given exception.
-     *
-     * @param e        The given Exception
-     * @param request  The http request
-     * @param response The http response
-     * @return A model view containing a error response object
-     */
-    @ExceptionHandler(Exception.class)
-    public ModelAndView handleException(final Exception e, final HttpServletRequest request,
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseRestApiControllerAdvice.class);
```

Comment:
Add error logging to each exception handler.  We should log the exception message and the stack trace:
LOGGER.error(e.getMessage(), e);

Also, review each exception handler and make sure to return a generic message instead of actual exception message as it might contain sensitive information.  For what to return refer to current implementation of RestErrorHandler.